src/cards/purify.ts

```diff
@@ -39,12 +41,18 @@ const spell: Spell = {
     },
   },
 };
-export function apply(unit: Unit.IUnit, underworld: Underworld) {
-
+export function apply(unit: Unit.IUnit, underworld: Underworld, prediction: boolean, sourceUnit: Unit.IUnit) {
+  let cursesRemoved = 0;
   for (let [modifier, modifierProperties] of Object.entries(unit.modifiers)) {
     if (modifierProperties.isCurse) {
       Unit.removeModifier(unit, modifier, underworld);
+      cursesRemoved += 1;
+      GameStatistics.trackCursePurified({ unit, sourceUnit }, underworld, prediction);
     }
   }
+
+  if (!prediction && sourceUnit == globalThis.player?.unit && cursesRemoved >= 5) {
```

Comment:
Rather than using the cursesRemoved variable on 3 lines, let's just do `unit.modifiers.filter(m => m.isCurse).length` here.  it keeps the logic more concise.